Content/NPCs/Misc/MagnetizedEnemies.cs

```diff
@@ -0,0 +1,278 @@
+
+        public Vector2 lightningDirection = Vector2.Zero;
+
+        public Vector2 endPoint = Vector2.Zero;
+
+        public Player chargedPlayer = default;
+
+        public Color baseColor = new(200, 230, 255);
+        public Color endColor = Color.Purple;
+
+        public List<Vector2> cache;
+        public List<Vector2> cache2;
+
+        public Trail trail;
+        public Trail trail2;
+
+        public float fade = 1f;
+
+
+        public override bool InstancePerEntity => true;
+
+        public override void OnSpawn(NPC npc, IEntitySource source)
+        {
+            if (npc.boss || npc.immortal || npc.dontTakeDamage || npc.friendly || npc.townNPC)
+                return;
+
+			if (!Main.rand.NextBool(50))
+				return;
+
+            if (source is EntitySource_SpawnNPC spawnSource)
+            {
+                if (Main.npc.Any(n => n.active && n.GetGlobalNPC<MagnetizedEnemies>().charged))
+                    return;
+
+                Player player = Main.player.Where(n => n.active && !n.dead).OrderBy(n => n.DistanceSQ(npc.Center)).FirstOrDefault();
+                if (player != default && player.HasItem(ModContent.ItemType<Items.Magnet.UnchargedMagnet>()))
+                {
+                    chargedPlayer = player;
+					charged = true;
+                    npc.lifeMax *= 4;
+                    npc.life = npc.lifeMax;
+                }
+            }
+        }
+
+        public override void AI(NPC npc)
+        {
+            if (charged && Main.rand.NextBool(6))
+            {
+                Vector2 dir = Main.rand.NextFloat(6.28f).ToRotationVector2();
+                Vector2 offset = Main.rand.NextBool(4) ? dir * Main.rand.NextFloat(30) : new Vector2(Main.rand.Next(-35, 35), npc.height / 2);
+
+                float smalLCharge = 0.5f;
+
+                var proj = Projectile.NewProjectileDirect(npc.GetSource_FromAI(), npc.Center + offset, dir.RotatedBy(Main.rand.NextFloat(-1, 1)) * 5, ModContent.ProjectileType<CloudstrikeShot>(), 0, 0, chargedPlayer.whoAmI, smalLCharge, 2);
+                var mp = proj.ModProjectile as CloudstrikeShot;
+                mp.velocityMult = Main.rand.Next(1, 4);
+
+                mp.thickness = 0.45f;
+                mp.host = npc;
+
+                mp.baseColor = Color.Cyan;
+            }
+
+            if (charged)
+            {
+                if (attackCounter++ % attackCycleLength < 200)
+                    lightningDirection = npc.DirectionTo(chargedPlayer.Center);
+
+                Vector2 offset = Vector2.Zero;
+
+                for (int k = 0; k < 50; k++)
+                {
+                    offset = lightningDirection * k * 16;
+
+                    int i = (int)((npc.Center.X + offset.X) / 16);
+                    int j = (int)((npc.Center.Y + offset.Y) / 16);
+                    Tile testTile = Main.tile[i, j];
+
+                    if (testTile.HasTile && Main.tileSolid[testTile.TileType] && !TileID.Sets.Platforms[testTile.TileType])
```

Comment:
single line conditional should be inlined

Reply:
fixed